packages/core/lib/testing/testrunner.js

```diff
@@ -196,11 +196,18 @@ TestRunner.prototype.endTest = function(mocha, callback) {
           return input.type;
         });
 
-        var values = abi.decodeLog(
-          event.abi_entry.inputs,
-          log.data,
-          log.topics.slice(1) // skip topic[0] for non-anonymous event
-        );
+        var values;
+        try {
+          values = abi.decodeLog(
+            event.abi_entry.inputs,
+            log.data,
+            log.topics.slice(1) // skip topic[0] for non-anonymous event
+          );
+        } catch (_) {
+          //temporary HACK until we're using the new decoder
+          self.logger.log(`    Warning: event decoding failed`);
```

Comment:
Maybe this should be something more informative to the user. I think maybe the message should be something that let's them know "Hey, the decoding failed but it is not that big a deal. This just happens in certain circumstances when you have similarly named events."

I'm not sure how to say that succinctly haha. Maybe
```
Warning: event decoding failed. 
This sometimes happens with events with duplicate names.
```

What do you think?

Reply:
I mean, that's kind of what "Warning" means, isn't it?  There are other reasons decoding can fail as well.

Note that this message doesn't actually come up in this case!  The problem here is in entirely in `contract`, not the test runner.  I just left this in here to be safe, basically?  <shrug>  I'm not very clear on when duplicate names cause problems and when they don't...

Reply:
This won't come up? If someone is using the `--show-events` flag then they will get the report at the end of the test runner feedback. I really just wanted to let people know that it is more a Truffle problem at this point rather than being something that they are doing wrong and should change. But again, if you disagree then don't let this hold up moving this forward.

Reply:
Sorry, when I say "in this case" I don't mean it won't come up at all (otherwise there'd be no point!), I mean it doesn't come up in the reported case that led to this.  But I didn't try it with `--show-events` so uhhh I guess maybe it comes up then, I should try that.

I just don't like splitting the message over two lines honestly, or suggesting a particular thing as the cause when there are multiple possible causes.  But maybe we can have something that fits on one line and still conveys the same idea.  E.g., "Event decoding failed due to internal Truffle error"?

Reply:
I'm really thinking about the fact that I think people are going to start reporting this to us after we implement the message. I was just thinking if we provided them with the explanation that it might "head them off at the pass" and they wouldn't need to report it if it was the case that they named their events with the same names.

Reply:
Oh, that's a good point.  I hadn't thought of that.  OK, yeah, I guess we do need to say something like that.  But I'll try to word it in a way that suggests the possibility of other causes. :)

Reply:
Sounds cool, either way I won't hold it back any more. Looks good

Reply:
Huh, testing it out with `--show-events` on, it indeed does not come up in this case.  I guess due to some difference between how `contract` and the test runner handle things.  Weird.  Well, in any case, I've updated the message.